server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/WriteLoadConstraintDeciderTests.java

```diff
@@ -117,6 +110,23 @@ public void testWriteLoadDeciderDisabled() {
      * Test the {@link WriteLoadConstraintDecider#canAllocate} implementation.
      */
     public void testWriteLoadDeciderCanAllocate() {
+        internalDeciderAllocationCheck((decider, shardRouting, node, allocation) -> decider.canAllocate(shardRouting, node, allocation));
+    }
+
+    /**
+     * Test the {@link WriteLoadConstraintDecider#canForceAllocateDuringReplace} implementation.
+     */
+    public void testWriteLoadDeciderCanForceAllocateDuringReplace() {
+        internalDeciderAllocationCheck(
+            (decider, shardRouting, node, allocation) -> decider.canForceAllocateDuringReplace(shardRouting, node, allocation)
+        );
+    }
+
+    interface DeciderAllocation {
```

Comment:
We don't have a style, of which I'm aware, for this sort of thing. I wonder if `DeciderCanAllocateOrForceAllocateCallback` would be a better name? It's essentially a means of supplying a lambda/callback to inject a choice from a higher level. I might have my terminology a little scrambled, but some indication in the name of purpose would make it easier to quickly understand.

Reply:
`DeciderInvocationStrategy` maybe?